app/controllers/upload.controller.js

```diff
@@ -251,10 +250,13 @@ exports.upload = async (req, res) => {
 							//console.error(`Error uploading chunk number ${e.id} - ${e.res.statusText}`);
 						});
 						uploader.on("done", (finishRes) => {
-							//console.log(`Upload completed with ID ${finishRes.data.id}`);
 							Upload.setHash(quoteId, index, finishRes.data.id);
 							// TODO: HEAD request to Arweave Gateway to verify that file uploaded successfully
-							Quote.setStatus(quoteId, Quote.QUOTE_STATUS_UPLOAD_END);
+
+							files_uploaded = files_uploaded + 1;
+							if(files_uploaded == files.length) {
+								Quote.setStatus(quoteId, Quote.QUOTE_STATUS_UPLOAD_END);
```

Comment:
I need to make sure that they all complete before setting it to UPLOAD_END. This corrects the bug I had when it would set it to UPLOAD_END after the first one succeeds.

I would like to check outside of the map(), but as in the above comment, I don't know how to change a variable outside of the block.